src/main/java/in/ac/bits/protocolanalyzer/persistence/repository/ElasticsearchConfig.java

```diff
@@ -49,7 +53,19 @@ public ElasticsearchOperations elasticsearchTemplate() {
                 .put("http.cors.allow-methods", allowMethods)
                 .put("http.cors.allow-headers", allowHeaders);
         NodeBuilder builder = NodeBuilder.nodeBuilder().local(true)
-                .settings(settingsBuilder.build());
+                .settings(settingsBuilder.build());*/
+	ImmutableSettings.Builder settingsBuilder = esFactoryImpl.settingsBuilder().put("cluster.name", clusterName)
+            .put("node.name", nodeName).put("node.data", true)
+	.put("path.data", dataPath).put("path.logs", logPath)
+            .put("index.number_of_shards", 1)
+            .put("index.number_of_replicas", 0)
+            .put("http.cors.enabled",
+                    Boolean.valueOf(corsEnabled).booleanValue())
+            .put("http.cors.allow-origin", allowOrigin)
+            .put("http.cors.allow-methods", allowMethods)
+            .put("http.cors.allow-headers", allowHeaders);
+	NodeBuilder builder = esFactoryImpl.nodeBuilder().local(true)
+			.settings(settingsBuilder.build());
         return new ElasticsearchTemplate(builder.node().client());
```

Comment:
@prasadtalasila We configure ElasticsearchTemplate by passing builder.node().client() into the constructor, but there is no getter defined in ElasticsearchTemplate to get this back and test it for mocked inputs.

Reply:
You can push the creation of this object into the abstract factory implementation. That way, the problem gets handled by the mocks.

Reply:
You can create create ESFactory.elasticsearchTemplate() method to handle the object returned in the last statement.